Utilities/Thread.cpp

```diff
@@ -1324,18 +1327,21 @@ bool handle_access_violation(u32 addr, bool is_writing, x64_context* context)
 				}
 			}
 		}
+	}
 
-		vm::temporary_unlock(*cpu);
-		LOG_FATAL(MEMORY, "Access violation %s location 0x%x", is_writing ? "writing" : "reading", addr);
-		cpu->state += cpu_flag::dbg_pause;
-		cpu->check_state();
+	if (cpu)
+	{
+		vm::passive_unlock(*cpu);
```

Comment:
Removed code was allowing to unpause thread in the debugger, and retry. Use this only as an alternative path (!cpu).